library/std/src/net/ip.rs

```diff
@@ -1724,43 +1740,6 @@ impl Ipv6Addr {
         }
     }
 
-    /// Converts this address to an [`IPv4` address]. Returns [`None`] if this address is
-    /// neither IPv4-compatible or IPv4-mapped.
-    ///
-    /// ::a.b.c.d and ::ffff:a.b.c.d become a.b.c.d
-    ///
-    /// [`IPv4` address]: Ipv4Addr
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use std::net::{Ipv4Addr, Ipv6Addr};
-    ///
-    /// assert_eq!(Ipv6Addr::new(0xff00, 0, 0, 0, 0, 0, 0, 0).to_ipv4(), None);
-    /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).to_ipv4(),
-    ///            Some(Ipv4Addr::new(192, 10, 2, 255)));
-    /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1).to_ipv4(),
-    ///            Some(Ipv4Addr::new(0, 0, 0, 1)));
-    /// ```
-    ///
-    /// # Stability guarantees
-    ///
-    /// This method's behavior [may by subject to changes][changes] in the
-    /// future, as new RFCs are published.
-    ///
-    /// [changes]: ../net/index.html#stability-guarantees-for-ietf-defined-behavior
-    #[rustc_const_unstable(feature = "const_ipv6", issue = "76205")]
-    #[stable(feature = "rust1", since = "1.0.0")]
-    pub const fn to_ipv4(&self) -> Option<Ipv4Addr> {
```

Comment:
This is a stable function, you can't delete it.

Reply:
yikes ok

Reply:
what about `to_ipv4_mapped` is that stable?

Reply:
because otherwise we're going to have two functions doing the same thing :'(

Reply:
it appears to not be in stable! https://github.com/rust-lang/rust/blob/master/library/std/src/net/ip.rs#L1495-L1503

Reply:
Ah hmm, it's weird that's missing a stability attribute, apparently that means it's [unstable by default](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2b485b88c6be8d134135f33e2f5a6694). So yeah, it seems fine to deprecate or remove, you probably want @KodrAus to give an opinion though.

Reply:
Ok. For the time being, I adjusted the behavior of `to_ipv4` to ignore IPv4-Compatible (the deprecated one) addresses and instead to only look for IPv4-Mapped (the currently-in-use-one) behavior. I kind of feel like I'm just wildly changing behavior so we should probably document this as a breaking change for checking for IPv4-Compatible addresses if this is the behavior we wind up going with.

Reply:
@jyn514 That would be the `#![unstable]` at the top of the module... Which catches me all the time 🙂

Reply:
It looks like we’ve made a few behavioral changes, but they’re all around consistency of mapped V4 addresses so I think we can roll that up into the stabilization here and let the rest of Libs weigh in on it.

Reply:
Another option for `to_ipv4` could be to deprecate it in favor of `to_ipv4_mapped` and stabilize that. That might be preferable here.